webapp/components/rhs_thread/rhs_thread.jsx

```diff
@@ -320,13 +322,23 @@ export default class RhsThread extends React.Component {
         return this.refs.postListContainer;
     }
 
+    handleClose() {
+        GlobalActions.emitCloseRightHandSide();
+        this.props.shrink();
+    }
+
     render() {
         if (this.props.posts == null || this.props.selected == null) {
             return (
                 <div/>
             );
         }
 
+        const isCurrentUserChannelMember = ChannelStore.isCurrentUserChannelMember(this.props.selected.channel_id);
+        if (!isCurrentUserChannelMember) {
+            this.handleClose();
```

Comment:
This is still calling an action as part of the render method. What we should be doing here is
1. Having the container pass in the channel member for this user and thread, if one exists. The ChannelStore method is convenient, but it doesn't fit with redux.
2. Instead of the render method, you can check whether or not `channelMember` has been removed in the `componentDidUpdate` method. It will have similar timing to doing it in render, but it'll be cleaner since this function is supposed to be pure (ie not change state or fire actions)